scene/gui/color_picker.cpp

```diff
@@ -901,12 +1008,24 @@ void ColorPicker::erase_preset(const Color &p_color) {
 			}
 		}
 
+		palette_name->set_text(vformat("%s*", palette_name->get_text().replace("*", "")));
+		palette_name->set_tooltip_text(ETR("The changes to this palette have not been saved to a file."));
+
 #ifdef TOOLS_ENABLED
 		if (editor_settings) {
 			PackedColorArray arr_to_save = get_presets();
 			editor_settings->call(SNAME("set_project_metadata"), "color_picker", "presets", arr_to_save);
+			editor_settings->call(SNAME("set_project_metadata"), "color_picker", "palette_edited", true);
+			if (preset_cache.is_empty()) {
+				palette_name->hide();
+				editor_settings->call(SNAME("set_project_metadata"), "color_picker", "palette_name", "");
+			}
 		}
 #endif
```

Comment:
I think this should be `#else`, otherwise the code below is called redundantly with the code above.